offload/liboffload/src/OffloadImpl.cpp

```diff
@@ -899,6 +899,15 @@ Error olCreateProgram_impl(ol_device_handle_t Device, const void *ProgData,
   return Error::success();
 }
 
+Error olIsValidBinary_impl(ol_device_handle_t Device, const void *ProgData,
+                           size_t ProgDataSize, bool *IsValid) {
+  // Make a copy of the program binary in case it is released by the caller.
```

Comment:
This can only happen if the caller frees it from another thread while we are calling this, right? I think that's an obnoxious enough thing to do that we should probably just call it UB. We have lots of other functions elsewhere that assume their arguments are not free'd during the call.